_compdemos/Cromwell.md

```diff
@@ -4,18 +4,101 @@ primary_reviewers: vortexing
+
+This is a valuable benefit because often when a complex or highly parallel workflow is run, there may be a variety of individual task failures due to problems that aren't in the code itself but rather things like internet connectivity problems or other intermittent problems.  Workflow managers can manage both the direct resubmission of individual failed tasks that failed due to HPC resource failures, as well as allow the user to restart an entire workflow should a task fail.  This means that work can be done more efficiently even when individual tasks might fail, because when the user might provide a fix or even directly resubmit a workflow, only those tasks impacted will be re-run, not those that completed successfully.  
 
+#### Using `Scratch` Space
+A real strength of using workflow managers is that your need to retain and archive large amounts of intermediate data involved in a workflow is substantially reduced and thus your data management and storage burdens are as well.  During your customization of your Cromwell server via our configuration, you can specify a location in `/fh/scratch` as your working directory for all your Cromwell workflows.  You can read more about `Scratch` storage at the Hutch [here.](/scicomputing/store_scratch/)  Cromwell will then create unique workflow- and task-level directories in that location for all of the input staging and task results.  Then, on a workflow-by-workflow level OR at the Cromwell server level, you can specify a location elsewhere (like in `Fast` or somewhere more long term) to store only the final workflow outputs specified. 
 
-## Design Recommendations for WDL workflows at Fred Hutch
-In order to improve sharability and also leverage the R package, as well as future UI based submission tools being developed, we recommend you structure your WDL based workflows with the following input files:
+In AWS, this is slightly different, but users specify prefixes in their [AWS S3 bucket](/scicomputing/store_objectstore/#economy-cloud-s3) where they want intermediate files to be saved, and then a prefix where they want their final outputs to go.  This allows us to then either manually or automatically (via help from `scicomp`), remove the intermediate files from S3 by deleting just a single prefix.  
+
+That allows users to use temporary, inexpensive storage for the workings of Cromwell, and then store their input data and their results in more long term storage.  This is critical for workflows that generate large amounts of intermediate data, like as commonly the case in genomics work, where a workflow may may 1-2TB of intermediate data, but the user may only want 500G of that data as their "final" outputs.  By being able to re-generate exactly any necessary intermediate file of a workflow, the need to retain and organize these data long term is eliminated.  
```

Comment:
This is critical for workflows that generate large amounts of intermediate data, which is commonly the case in genomics work. A workflow may produce 1-2TB of intermediate data, but the user may only want 500G of that data as their "final" outputs.